src/utils.js

```diff
@@ -81,14 +81,23 @@ const patchIdsWithParentId = (schema, parentId) => {
       }
     : {
         ...schema,
-        properties: properties.reduce((memo, [key, prop]) => {
-          return {
-            ...memo,
-            [key]: {
-              ...prop,
-              $id: `${parentId}/${prop.$id.replace('#', '')}`, // e.g. #properties/foo/properties/bar
-            },
-          }
+        properties: properties.reduce((memo, [key, props]) => {
+          return !props.$id
+            ? {
+                ...memo,
+                [key]: {
+                  ...props,
+                },
+              }
+            : {
+                ...memo,
+                [key]: {
+                  ...props,
+                  $id: generateIds
+                    ? `${parentId}/${props.$id.replace('#', '')}`
+                    : props.$id, // e.g. #properties/foo/properties/bar
+                },
+              }
```

Comment:
I would argue that an if-then-else is more readable than a 16-lines ternary.

Reply:
I like having only a return when it's possible but you are right it doesn't look nice.